### PR TITLE
[Anomaly Detector][Synapse] Update Endpoint type from string to Uri

### LINKED ISSUE #19225: Update Endpoint type from string to Uri
* Update [Anomaly Detector](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/anomalydetector/Azure.AI.AnomalyDetector/src/Generated/AnomalyDetectorClient.cs#L34)
* Update 
  - [Synapse.AccessControl](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/synapse/Azure.Analytics.Synapse.AccessControl/src/Generated/RoleDefinitionsClient.cs#L35)
  - [Synapse.Artifacts](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/synapse/Azure.Analytics.Synapse.Artifacts/src/Generated/BigDataPoolsClient.cs#L34)
  - [Synapse.ManagedPrivateEndpoints](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/synapse/Azure.Analytics.Synapse.ManagedPrivateEndpoints/src/Generated/ManagedPrivateEndpointsClient.cs#L34)
  - [Synapse.Monitoring](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/synapse/Azure.Analytics.Synapse.Monitoring/src/Generated/MonitoringClient.cs#L34)
  - [Synapse.Spark](https://github.com/Azure/azure-sdk-for-net/blob/master/sdk/synapse/Azure.Analytics.Synapse.Spark/src/Generated/SparkBatchClient.cs#L36)

### PR DESCRIPTION
**Pull request by ShivangiReja**
Fixes: https://github.com/Azure/azure-sdk-for-net/issues/19225

### CODE REVIEW

**Comment by pakrym** [comment]
Hm, why didn't any piepline run?

File: sdk/synapse/Azure.Analytics.Synapse.Spark/api/Azure.Analytics.Synapse.Spark.netstandard2.0.cs [hunk 1]
```diff
@@ -3,7 +3,7 @@ namespace Azure.Analytics.Synapse.Spark
     public partial class SparkBatchClient
     {
         protected SparkBatchClient() { }
-        public SparkBatchClient(string endpoint, string sparkPoolName, Azure.Core.TokenCredential credential, string livyApiVersion = "2019-11-01-preview", Azure.Analytics.Synapse.Spark.SparkClientOptions options = null) { }
+        public SparkBatchClient(System.Uri endpoint, string sparkPoolName, Azure.Core.TokenCredential credential, string livyApiVersion = "2019-11-01-preview", Azure.Analytics.Synapse.Spark.SparkClientOptions options = null) { }
```
Comment (pakrym) [inline sdk/synapse/Azure.Analytics.Synapse.Spark/api/Azure.Analytics.Synapse.Spark.netstandard2.0.cs:6]: We might need to fix livyApiVersion eventually

**Review by pakrym — APPROVED** [review]

File: sdk/synapse/Azure.Analytics.Synapse.Spark/api/Azure.Analytics.Synapse.Spark.netstandard2.0.cs (hunk 1 above)
Comment (ShivangiReja) [reply sdk/synapse/Azure.Analytics.Synapse.Spark/api/Azure.Analytics.Synapse.Spark.netstandard2.0.cs:6]: CodeModel.yaml is no setting `origin: modelerfour:synthesized/api-version` for [livyApiVersion](https://github.com/Azure/azure-rest-api-specs/blob/aa19725fe79aea2a9dc580f3c66f77f89cc34563/specification/synapse/data-plane/Microsoft.Synapse/preview/2019-11-01-preview/sparkJob.json#L1483), that's why this param is here. Something is now right in the swagger for api-version.

**Comment by ShivangiReja** [comment]
@pakrym Accidentally deleted your comment.

>@pakrym We might need to fix livyApiVersion eventually

CodeModel.yaml is no setting `origin: modelerfour:synthesized/api-version` for [livyApiVersion](https://github.com/Azure/azure-rest-api-specs/blob/aa19725fe79aea2a9dc580f3c66f77f89cc34563/specification/synapse/data-plane/Microsoft.Synapse/preview/2019-11-01-preview/sparkJob.json#L1483), that's why this param is here. Something is not right in the swagger for api-version.

File: sdk/synapse/Azure.Analytics.Synapse.Spark/api/Azure.Analytics.Synapse.Spark.netstandard2.0.cs (hunk 1 above)
Comment (ShivangiReja) [inline sdk/synapse/Azure.Analytics.Synapse.Spark/api/Azure.Analytics.Synapse.Spark.netstandard2.0.cs:6]: @pakrym Accidentally deleted your comment.

> @pakrym We might need to fix livyApiVersion eventually

Comment (ShivangiReja) [reply sdk/synapse/Azure.Analytics.Synapse.Spark/api/Azure.Analytics.Synapse.Spark.netstandard2.0.cs:6]: CodeModel.yaml is no setting `origin: modelerfour:synthesized/api-version` for [livyApiVersion](https://github.com/Azure/azure-rest-api-specs/blob/aa19725fe79aea2a9dc580f3c66f77f89cc34563/specification/synapse/data-plane/Microsoft.Synapse/preview/2019-11-01-preview/sparkJob.json#L1483), that's why this param is here. Something is not right in the swagger for api-version.

**Pull request merged**